doc/bundle_lookup_matcher.md

```diff
@@ -0,0 +1,132 @@
+
+A good observer may notice that loading both `main/en/ca-gregorian` and `main/en-US/ca-gregorian` is redundant. Although loading both is not a problem, loading either the `en` or the `en-US` bundle alone is enough.
+
+Let's add a bit of sugar to the requested locales.
+
+```javascript
+var en = new Cldr( "en" );
+console.log( en.attributes.bundle ); // "en"
+
+var enUs = new Cldr( "en-US" );
+console.log( enUs.attributes.bundle ); // "en"
+
+var enLatnUs = new Cldr( "en-Latn-US" );
+console.log( enLatnUs.attributes.bundle ); // "en"
+```
+
+All instances above obviously matches the same `en` bundle. Because, (a) `en` is the default content for `en-US` and (b) `en-US` is the default content for `en-Latn-US`.
+
+What happens if the requested locale includes unicode extensions?
+
+```javascript
+var en = new Cldr( "en-US-u-cu-USD" );
+console.log( en.attributes.bundle ); // "en"
+console.log( en.main( "numbers/currencies/{u-cu}/displayName" ) ); // "US Dollar"
+```
+
+Unicode extensions are obviously ignored on bundle lookup. Note they are accessible via variable replacements.
+
+Follow other non-obvious lookups.
+
+```javascript
+Cldr.load(
+  cldr( "supplemental/likelySubtags" ),   // JSON data from supplemental/likelySubtags.json
+  cldr( "main/sr-Cyrl/numbers" ),         // JSON data from main/sr-Cyrl/numbers.json
+  cldr( "main/sr-Latn/numbers" ),         // JSON data from main/sr-Latn/numbers.json
+  cldr( "main/zh-Hant/numbers" )          // JSON data from main/zh-Hant/numbers.json
+);
+
+var srCyrl = new Cldr( "sr-Cyrl" );
+console.log( srCyrl.attributes.bundle ); // "sr-Cyrl"
+console.log( srCyrl.main( "numbers/decimalFormats-numberSystem-latn/short/decimalFormat/1000-count-one" ) );
+// ➜ "0 хиљ'.'"
+
+var srRS = new Cldr( "sr-RS" );
+console.log( srRs.attributes.bundle ); // "sr-Cyrl"
+console.log( srRs.main( "numbers/decimalFormats-numberSystem-latn/short/decimalFormat/1000-count-one" ) );
+// ➜ "0 хиљ'.'"
+
+var srLatnRS = new Cldr( "sr-Latn-RS" );
+console.log( srLatnRS.attributes.bundle ); // "sr-Latn"
+console.log( srLatnRS.main( "numbers/decimalFormats-numberSystem-latn/short/decimalFormat/1000-count-one" ) );
+// ➜ "0 hilj'.'"
+
+var zhTW = new Cldr( "zh-TW" );
+console.log( zhTW.attributes.bundle ); // "zh-Hant"
+console.log( zhTW.main( "numbers/symbols-numberSystem-hanidec/nan" ) ); // "非數值"
+```
+
+Finally, if an instance is created whose bundle hasn't been loaded yet, its `.attributes.bundle` is set as `undefined`. If this instance is used to traverse a main dataset, an error is thrown.
+
+```javascript
+var zhCN = new Cldr( "zh-CN" );
+console.log( zhCN.attributes.bundle ); // undefined
+console.log( zhCN.main( /* something */ ) ); // Error: E_MISSING_BUNDLE
```

Comment:
This makes me wonder why the error isn't thrown when calling the constructor. I guess there is a reason, might be good to include it here.